examples/pytorch/text-classification/run_glue.py

```diff
@@ -507,17 +507,25 @@ def compute_metrics(p: EvalPrediction):
         if data_args.task_name == "mnli":
             tasks.append("mnli-mm")
             eval_datasets.append(raw_datasets["validation_mismatched"])
+            combined = {}
 
         for eval_dataset, task in zip(eval_datasets, tasks):
             metrics = trainer.evaluate(eval_dataset=eval_dataset)
 
             max_eval_samples = (
-                data_args.max_eval_samples if data_args.max_eval_samples is not None else len(eval_dataset)
+                data_args.max_eval_samples
+                if data_args.max_eval_samples is not None
+                else len(eval_dataset)
             )
             metrics["eval_samples"] = min(max_eval_samples, len(eval_dataset))
 
+            if task == "mnli-mm":
+                metrics = {k + "_mm": v for k, v in metrics.items()}
+            if "mnli" in task:
+                combined.update(metrics)
```

Comment:
I'd regroup this slightly differently:
```suggestion
            if task == "mnli-mm":
                combined.update({k + "_mm": v for k, v in metrics.items()})
            elif task == "mnli":
                combined.update(metrics)
```

Reply:
With this regrouping, the `trainer.log_metrics("eval", metrics)` won't have the `"_mm"` , so the stdout will be the same as before (that's the main reason why keys in `metrics` are updated). Given this, should we still regroup and affect only the json log files, but not the stdout?